Core/Services/OfferService.cs

```diff
@@ -44,14 +59,39 @@ public async Task<OfferCreateDTO> CreateOfferAsync(OfferCreateDTO offerCreate, s
             offer.OfferCreatorId = userId;
             offer.CreationDate = DateTimeOffset.UtcNow;
             offer.IsClosed = false;
-            offer.CreatorRoleId = _roleRepository.GetRoleByName(offerCreate.Role);
-            offer.GoodCategoryId = _goodCategoryRepository
+            offer.CreatorRoleId = _roleService.GetRoleByName(offerCreate.Role);
+            offer.GoodCategoryId = _goodCategoryService
                 .GetGoodCategoryByName(offerCreate.GoodCategory);
             offer.OfferPointId = await _pointService.CreatePointForOfferAsync(offerCreate.Point);
 
             await _offerRepository.InsertAsync(offer);
             await _offerRepository.SaveChangesAsync();
             return _mapper.Map<OfferCreateDTO>(offerCreate);
         }
+
+        public async Task<OfferInfoDTO> GetConcreteOfferAsync(int offerId, string userId)
+        {
+            var offer = await _offerRepository.GetByIdAsync(offerId);
+
+            ExceptionMethods.OfferNullCheck(offer);
+            ExceptionMethods.UserNullCheck(await _userManager.FindByIdAsync(userId));
+
+            if (offer.OfferCreatorId != userId)
+            {
+                throw new HttpException(ErrorMessages.OfferNotFound, HttpStatusCode.NotFound);
+            }
```

Comment:
Are you sure that BL should know anything about web UI ?

Reply:
What if we get a request not from web UI, how do we check it?